runtime/lua/vim/treesitter/dev.lua

```diff
@@ -501,12 +501,19 @@ function M.inspect_tree(opts)
     end,
   })
 
-  api.nvim_create_autocmd('BufHidden', {
+  api.nvim_create_autocmd({ 'WinClosed' }, {
     group = group,
     buffer = buf,
-    once = true,
     callback = function()
-      close_win(w)
+      local windows = vim.fn.win_findbuf(buf) --[=[@as integer[]]=]
+      if #windows == 1 then
+        close_win(w)
+        return true
+      end
+
+      win = vim.iter(windows):find(function(new_win)
```

Comment:
This is somewhat subtle (updating shared state that is "far" from this scope). So worth a comment.
```suggestion
      -- Maintain a valid window, if the original window was closed.
      win = vim.iter(windows):find(function(new_win)
```